modules/Classroom/components/MessageList/MessageList.tsx

```diff
@@ -1,15 +1,49 @@
+import { useEffect, useState } from "react";
+
 import { Box, Text } from "@mantine/core";
-import dayjs from "dayjs";
+import { io, Socket } from "socket.io-client";
 
+import { WS_BASE_URL } from "@/shared/constants/env.constants";
+import { useAppSelector } from "@/shared/redux/hooks";
+import { selectAuthenticatedUser } from "@/shared/redux/reducers/user.reducer";
 import { useGetClassroomMessagesQuery } from "@/shared/redux/rtk-apis/messages/messages.api";
+import { IMessage } from "@/shared/redux/rtk-apis/messages/messages.interface";
 
 import MessageCard from "../MessageCard/MessageCard";
 import { IMessageListProps } from "./MessageList.interface";
 import useStyles from "./MessageList.styles";
 
 const MessageList: React.FC<IMessageListProps> = ({ classroomId }) => {
-  const { data: messages } = useGetClassroomMessagesQuery(classroomId);
+  const { data: initialMessages, isSuccess, refetch } = useGetClassroomMessagesQuery(classroomId);
   const { classes } = useStyles();
+  const [messages, setMessages] = useState<IMessage[]>([]);
+  const user = useAppSelector(selectAuthenticatedUser);
+
+  useEffect(() => {
+    refetch();
+  }, [classroomId, refetch]);
+
+  useEffect(() => {
+    if (isSuccess && initialMessages) {
+      setMessages(initialMessages);
+    }
+  }, [initialMessages, isSuccess]);
+
+  useEffect(() => {
+    const socket: Socket = io(WS_BASE_URL);
+
+    socket.emit("joinClassroom", { classroomId, userId: user.userId });
+
+    socket.on("receiveMessage", (newMessage) => {
+      setMessages((prevMessages) => [newMessage, ...prevMessages]);
+    });
+
+    return () => {
+      socket.emit("leaveClassroom", { classroomId, userId: user.userId });
+      socket.off("receiveMessage");
+      socket.disconnect();
```

Comment:
```suggestion
      socket.disconnect();
```

Wouldn't this implicitly take care of the rest?